### PR TITLE
Feature/#199 add an endpoint to configure languages a student can use

### PR DESCRIPTION
**Pull request by begoIT24**
POST /student/{studentId}/resume/languages
Add a selected language from the languages table to the student's resume: corresponds to the plus "+" button in this view

image

### CODE REVIEW

**Review by ivanlegranbizarro — APPROVED** [review]

File: app/Http/Requests/AddStudentLanguageRequest.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?php
+
```
Comment (jordimorillo) [inline app/Http/Requests/AddStudentLanguageRequest.php:2]: declare(strict_types=1);

File: routes/api/v1.php [hunk 2]
```diff
@@ -16,14 +16,16 @@
...
+    AddStudentLanguageController
+};
 
 use App\Http\Controllers\api\Tag\{
     TagListController,
     TagStoreController,
     TagDetailController,
     TagUpdateController,
-    DevelopmentListController
+    DevelopmentListController
```
Comment (jordimorillo) [inline routes/api/v1.php:28]: There should be no blank spaces at the end

File: routes/api/v1.php [hunk 3]
```diff
@@ -48,6 +50,7 @@
     Route::put('projects/{projectId}', UpdateStudentProjectController::class)->name('student.updateproject');
     Route::put('skills', UpdateStudentSkillsController::class)->name('student.skills');
     Route::put('profile', UpdateStudentProfileController::class)->name('student.updateProfile');
+    Route::post('languages', AddStudentLanguageController::class)->name('student.addlanguage');
```
Comment (jordimorillo) [inline routes/api/v1.php:53]: To keep coherence with the rest of the code that name should be addLanguage in camel case.

File: tests/Feature/Controller/Student/AddStudentLanguageControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        ]);
+
+        $response->assertStatus(404);
+
+        $response->assertJson(['message' => 'No s\'ha trobat cap currículum per a l\'estudiant amb id: ' . $this->student->id]);
+    }
+
+    public function testAddStudentLanguageControllerReturns422ForNonExistentLanguageUuid(): void
+    {
+        $noExistentLanguageId = 'ab9bb2ed-8bb5-4a3a-bdb2-09cd00000f0b';
```
Comment (jordimorillo) [inline tests/Feature/Controller/Student/AddStudentLanguageControllerTest.php:74]: Extract this as a constant. In example:

self::NON_EXISTENT_LANGUAGE_ID

File: tests/Feature/Controller/Student/AddStudentLanguageControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+                'language_id' => [
+                    'Language id és invàlid.'
+                ]
+            ]
+        ]);
+    }
+
+    public function testAddStudentLanguageControllerReturns422ForInvalidLanguageUuid(): void
+    {
+        $invalidLanguageId = 'invalidLanguageId';
```
Comment (jordimorillo) [inline tests/Feature/Controller/Student/AddStudentLanguageControllerTest.php:93]: Same. Extract as a constant.

File: tests/Feature/Service/Student/AddStudentLanguageServiceTest.php [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+            'resume_id' => $this->resume->id,
+            'language_id' => $this->language->id,
+        ]);
+    }
+
+    public function testExecuteThrowsStudentNotFoundException(): void
+    {
+        $this->expectException(StudentNotFoundException::class);
+
+        $invalidStudentId = 'invalidStudentId';
```
Comment (jordimorillo) [inline tests/Feature/Service/Student/AddStudentLanguageServiceTest.php:55]: Review the code and extract the remaining constants. If they are frequently used extract a fixture file to keep them together and use them as public constants in your tests.

**Review by jordimorillo — APPROVED** [review]
Very good code. Please check those little observations.

File: routes/api/v1.php (hunk 2 above)
Comment (begoIT24) [reply routes/api/v1.php]: done

File: routes/api/v1.php (hunk 3 above)
Comment (begoIT24) [reply routes/api/v1.php]: done

File: tests/Feature/Controller/Student/AddStudentLanguageControllerTest.php (hunk 4 above)
Comment (begoIT24) [reply tests/Feature/Controller/Student/AddStudentLanguageControllerTest.php]: done for INVALID_STUDENT_ID, NON_EXISTENT_LANGUAGE_ID and INVALID_LANGUAGE_ID

File: tests/Feature/Controller/Student/AddStudentLanguageControllerTest.php (hunk 5 above)
Comment (begoIT24) [reply tests/Feature/Controller/Student/AddStudentLanguageControllerTest.php]: done

File: tests/Feature/Service/Student/AddStudentLanguageServiceTest.php (hunk 6 above)
Comment (begoIT24) [reply tests/Feature/Service/Student/AddStudentLanguageServiceTest.php:55]: all done

**Pull request merged**